tests/integration/test_charm_one_runner.py

```diff
@@ -239,7 +239,11 @@ async def test_token_config_changed_insufficient_perms(
     """
     unit = app.units[0]
 
-    await app.set_config({"token": "", "virtual-machines": "0"})
+    await app.set_config({"token": "invalid-token", "virtual-machines": "0"})
     await model.wait_for_idle()
 
     await wait_till_num_of_runners(unit, num=0)
+
+    # Add the runner back for the next test using the `app` deployment. Not part of the testing.
+    await app.set_config({"token": token, "virtual-machines": "1"})
+    await model.wait_for_idle(status=ACTIVE, timeout=10 * 60)
```

Comment:
Imho, this should be part of a setup/teardown fixture, see for example `test_charm_with_proxy` which has a setup fixture called `app` to ensure that the number of runners is zero.